src/test/scala/org/broadinstitute/dsde/firecloud/dataaccess/MockThurloeDAO.scala

```diff
@@ -56,6 +56,7 @@ class MockThurloeDAO extends ThurloeDAO {
   var mockKeyValues: Map[String, Set[FireCloudKeyValue]] =
     Map(
       NORMAL_USER -> baseProfile,
+      "foo" -> baseProfile, // to match registeredUser, enabledUserInfo, unknownUserInfo from MockSamDao
```

Comment:
the fix to save registered user's profiles under `userInfo.userSubjectId` requires this tweak to the testing mock